public/docs/dart/latest/guide/dependency-injection.jade

```diff
@@ -1,6 +1,782 @@
+
+  What if the `Engine` class evolves and its constructor requires a parameter?
+  Our `Car` is broken and stays broken until we rewrite it along the lines of
+  `this.engine = new Engine(theNewParameter)`.
+  We didn't care about `Engine` constructor parameters when we first wrote `Car`.
+  We don't really care about them now.
+  But we'll *have* to start caring because
+  when the definion of `Engine` changes, our `Car` class must change.
+  That makes `Car` brittle.
+
+  What if we want to put a different brand of tires on our `Car`. Too bad.
+  We're locked into whatever brand the `Tires` class creates. That makes our `Car` inflexible.
+
+  Right now each new car gets its own engine. It can't share an engine with other cars.
+  While that makes sense for an automobile engine,
+  we can think of other dependencies that should be shared ... like the onboard
+  wireless connection to the manufacturer's service center. Our `Car` lacks the flexibility
+  to share services that have been created previously for other consumers.
+
+  When we write tests for our `Car` we're at the mercy of its hidden dependencies.
+  Is it even possible to create a new `Engine` in a test environment?
+  What does `Engine`itself depend upon? What does that dependency depend on?
+  Will a new instance of `Engine` make an asynchronous call to the server?
+  We certainly don't want that going on during our tests.
+
+  What if our `Car` should flash a warning signal when tire pressure is low.
+  How do we confirm that it actually does flash a warning
+  if we can't swap in low-pressure tires during the test?
+
+  We have no control over the car's hidden dependencies.
+  When we can't control the dependencies, a class become difficult to test.
+
+  How can we make `Car` more robust, more flexible, and more testable?
+
+  That's super easy. We probably already know what to do. We change our `Car` constructor to this:
+
+  <a id="ctor-injection"></a>
+
++makeTabs(
+'dependency-injection/dart/web/car/car.dart, dependency-injection/dart/web/car/car-no-di.dart',
+'car-ctor, car-ctor',
+'bin/car/car.dart DI (constructor), bin/car/car.dart No DI (constructor)')(format=".")
+
+:marked
+  See what happened? We moved the definition of the dependencies to the constructor.
+  Our `Car` class no longer creates an engine or tires.
+  It just consumes them.
+
+.l-sub-section
+  :marked
+    We also leverage Dart's constructor syntax for declaring parameters and properties simultaneously.
+:marked
+  Now we create a car by passing the engine and tires to the constructor.
+
++makeExample('dependency-injection/dart/web/car/car-creations.dart', 'car-ctor-instantiation', 'Simple Car')(format=".")
+
+:marked
+  How cool is that?
+  The definition of the engine and tire dependencies are decoupled from the `Car` class itself.
+  We can pass in any kind of engine or tires we like, as long as they
+  conform to the general API requirements of an engine or tires.
```

Comment:
This is more strict in Dart and might need another phrase. "implements the same interface" or similar.